pixeltable/io/external_store.py

```diff
@@ -50,32 +69,106 @@ class Project(ExternalStore, abc.ABC):
+                # duplication. Otherwise, we'll create a new one.
+                for store in tbl_version.external_stores.values():
+                    if isinstance(store, Project) and col in store.stored_proxies:
+                        self.stored_proxies[col] = store.stored_proxies[col]
+                        break
+                if col not in self.stored_proxies:
+                    # We didn't find it in an existing Project
+                    stored_proxies_needed.append(col)
+        if len(stored_proxies_needed) > 0:
+            _logger.info(f'Creating stored proxies for columns: {[col.name for col in stored_proxies_needed]}')
+            # Create stored proxies for columns that need one. Increment the schema version
+            # accordingly.
+            tbl_version.version += 1
+            preceding_schema_version = tbl_version.schema_version
+            tbl_version.schema_version = tbl_version.version
+            proxy_cols = [self.create_stored_proxy(tbl_version, col) for col in stored_proxies_needed]
+            # Add the columns; this will also update table metadata.
+            tbl_version._add_columns(proxy_cols, conn)
+            # We don't need to retain `UpdateStatus` since the stored proxies are intended to be
+            # invisible to the user.
+            tbl_version._update_md(time.time(), conn, preceding_schema_version=preceding_schema_version)
+
+    def unlink(self, tbl_version: TableVersion, conn: sql.Connection) -> None:
+        # Determine which stored proxies can be deleted. (A stored proxy can be deleted if it is not referenced by
+        # any *other* external store for this table.)
+        deletions_needed: set[Column] = set(self.stored_proxies.values())
+        for name, store in tbl_version.external_stores.items():
+            if name != self.name:
+                deletions_needed = deletions_needed.difference(set(store.stored_proxies.values()))
+        if len(deletions_needed) > 0:
+            _logger.info(f'Removing stored proxies for columns: {[col.name for col in deletions_needed]}')
+            # Delete stored proxies that are no longer needed.
+            tbl_version.version += 1
+            preceding_schema_version = tbl_version.schema_version
+            tbl_version.schema_version = tbl_version.version
+            tbl_version._drop_columns(deletions_needed)
+            self.stored_proxies.clear()
+            tbl_version._update_md(time.time(), conn, preceding_schema_version=preceding_schema_version)
+
+    def create_stored_proxy(self, tbl_version: TableVersion, col: Column) -> Column:
+        from pixeltable import exprs
+
+        assert col.col_type.is_media_type() and not (col.is_stored and col.compute_func) and col not in self.stored_proxies
+        proxy_col = Column(
+            name=None,
+            # Force images in the proxy column to be materialized inside the media store,
```

Comment:
Point out that this is a hack and that the correct solution is the destination property in general (not just for some media types).

The todo is to remove this and replace with a simple columnref.